.github/workflows/ci.yml

```diff
@@ -0,0 +1,117 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  # XXX: macOS
+  ubuntu:
+    strategy:
+      matrix:
+          version: ['7.3', '7.4', '8.0', '8.1']
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Cache DB2 library
+        id: cache-clidriver
+        uses: actions/cache@v2
+        with:
+          path: clidriver
+          key: ${{ runner.os }}-clidriver
+      - name: Install DB2 library
+        if: steps.cache-clidriver.outputs.cache-hit != 'true'
+        run: |
+          wget https://public.dhe.ibm.com/ibmdl/export/pub/software/data/db2/drivers/odbc_cli/linuxx64_odbc_cli.tar.gz
+          tar xvzf linuxx64_odbc_cli.tar.gz
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{matrix.version}}
+          extensions: pdo
+      - name: phpize
+        run: phpize
+      - name: configure
+        run: ./configure --with-pdo-ibm=$PWD/clidriver
+      - name: make
+        run: make V=1
+      - name: Set up Db2 LUW in Docker
+        # XXX: Should we be caching the Docker image? Are we creating the necessary things?
+        # Adapted from the Travis setup with the changes used for the current
+        # version of the Db2 container.
+        run: |
+          set -x
+          cat <<EOF > db2cli.ini
+          [SAMPLE]
+          Hostname=localhost
+          Protocol=TCPIP
+          Port=60000
+          Database=sample
+          EOF
+          mkdir database
+          docker pull ibmcom/db2
+          docker run --name db2 --privileged=true -p 60000:50000 -e DB2INST1_PASSWORD=password -e LICENSE=accept -e DBNAME=sample -v database:/database -itd ibmcom/db2
+          docker ps -as
+          sleep 120 # XXX: NOT CORRECT, block on the docker run instead, somehow
```

Comment:
FYI, this is what I did for ibm_db: https://github.com/ibmdb/python-ibmdb/blob/master/scripts/setup-docker.sh#L12